### ISSUE TITLE
require-await rule creates unresolvable error

### ISSUE THREAD

**Issue by EdJoPaTo** [xojs/eslint-config-xo-typescript#12]
2a6857de7be2d3c18e8a8b91bfa25c683fbf3cd9 introduces a problem:
Either way I fix the rule, the other denies the code:

```ts
const results = await Promise.all(
	queries.map(await o => search(o))
);
```

Errors:
- Functions that return promises must be async. `@typescript-eslint/promise-function-async`
- Async arrow function has no await expression. `@typescript-eslint/require-await`

I personally prefer to have the async at the arrow function as it directly shows 'this returns a promise, it has to be dealt with'.

**Comment by sindresorhus** [comment]
I think this will fix it: https://github.com/typescript-eslint/typescript-eslint/pull/826

**Comment by EdJoPaTo** [comment]
The PR is in master but it is definitely not fine. I had no time to add PRs to the eslint Repo about what specifically does not work. But there are multiple different things not working.
I disabled the rule entirely for now as it has too many false positives.
